### ISSUE TITLE
[typescript] Cannot use TextField as a child of Field

### ISSUE THREAD

**Issue by krzysztof1980** [stackworx/formik-material-ui#49]
Hi, in KitchenSink.story.tsx and other examples TextField is passed as a value of component-property:
```typescript
<Field type="email" label="Email" name="email" component={TextField} />
```
I would like to use TextField as a child of Field, because then its properties should be recognized by the compiler, as they have a concrete type, unlike any in case of Field:
```typescript
<Field name="email">
    {(fieldProps: FieldProps) => (
        <TextField {...fieldProps} type="email" label="Email" />
    )}
</Field>
```
If I make this change, then it produces following error:
```
Error:(57, 24) TS2740: Type '{ type: string; label: string; field: { onChange: (e: ChangeEvent<any>) => void; onBlur: (e: any) => void; value: any; name: string; }; form: FormikProps<any>; }' is missing the following properties from type 'TextFieldProps': style, classes, innerRef, className, and 7 more.
```
Follwing changes seem to fix it (and don't brake any tests):
- in TextField.tsx change TextFieldProps from interface to a type:
```typescript
export type TextFieldProps = FieldProps & Omit<MuiTextFieldProps, 'error' | 'name' | 'onChange' | 'value'>
```
- in types.ts change the definition of Omit to the one implemented by material-ui project:
```typescript
export type Omit<T, K extends keyof any> = T extends any ? Pick<T, Exclude<keyof T, K>> : never;
```
With this solution the properties of Material-UI's TextField get recognized and I get Intellisense support from my IDE. I suspect similar problem can have another components, but I haven't time to test it. The solution used in storybook, when the component is passed as a prop, is shorter, but then the component's props are not type checked.

**Comment by cliedeman** [comment]
This does not seem to work anymore. even the variant I got specifically for a union types :/

**Comment by cliedeman** [comment]
When I get around to it again I will explicitly export the different TextField variants. This should take care of these type issues.

E.g OutlinedTextField, StandardTextField, FilledTextField

It's a bit clunky but the typechecking should be worth it

Ciaran

**Comment by cliedeman** [comment]
I think this has been fixed by #97
